app/models/proxy.rb

```diff
@@ -609,12 +608,26 @@ def policies_configs_are_correct
     end
   end
 
+  def read_and_parse_policies_config(raise_error: false)
+    attr_policies_config = read_attribute(:policies_config)
+    attr_policies_config.blank? ? [] : Array(JSON.parse(attr_policies_config))
+  rescue JSON::ParserError => error
+    raise_error ? raise(error) : []
+  end
+
+  def read_and_parse_policies_config!
+    read_and_parse_policies_config(raise_error: true)
+  end
```

Comment:
Imo the simple fact of having one bang method and one non-bang method should already imply raising the error or not, without that extra param.

So I propose something like this:

```suggestion
  def read_and_parse_policies_config
    read_and_parse_policies_config!
  rescue JSON::ParserError
    []
  end

  def read_and_parse_policies_config!
    attr_policies_config = read_attribute(:policies_config)
    attr_policies_config.blank? ? [] : Array(JSON.parse(attr_policies_config))
  end
```

Reply:
ok, that's super nice!